data/recs/2009/tsv_maker_2009.py

```diff
@@ -0,0 +1,292 @@
+
+    def add_missing_dependency_rows(self, df, project, count, weight):
+        """
+        Add combinations of dependencies for which we have no sample data; distribute option probabilities for these rows uniformly.
+
+        Parameters:
+          df (dataframe): A pandas dataframe with dependency/option columns and fractions.
+          project (str): Name of the project.
+          count (dataframe): A pandas dataframe with dependency columns and source sample sizes.
+          weight (dataframe): A pandas dataframe with dependency columns and source weights.
+
+        Returns:
+          df (dataframe): A pandas dataframe updated with missing dependency rows, source sample sizes, and source weights.
+
+        """
+        levels = df.index.levels
+        names = df.index.names
+        option_cols = df.columns.values
+        df = df.reset_index()
+
+        if 'testing' in project:
+            df = df[0: 0]
+
+        for group in itertools.product(*levels):
+            if not group in list(df.groupby(names).groups):
+                data = dict(zip(names, group))
+                data.update(
+                    dict(zip(option_cols, [1.0 / len(option_cols)] * len(option_cols))))
+                df = df.append(data, ignore_index=True, verify_integrity=True)
+        df = df.merge(count, on=names, how='outer')
+        df = df.merge(weight, on=names, how='outer')
+        df = df.fillna(0)
+        return df
+
+    def rename_cols(self, df, dependency_cols, project):
+        """
+        Prepend 'Dependency=' to dependency columns and 'Option=' to option columns. Sort by levels in order of dependencies from left to right.
+
+        Parameters:
+          df (dataframe): A pandas dataframe with dependency/option columns and fractions.
+          dependency_cols (list): A list of the dependency column names.
+          project (str): Name of the project.
+
+        Returns:
+          df (dataframe): A pandas dataframe with updated dependency/column names.
+
+        """
+        new_dependency_cols = []
+        for dependency_col in dependency_cols:
+            new_dependency_col = 'Dependency={}'.format(dependency_col)
+            df = df.rename(columns={dependency_col: new_dependency_col})
+            new_dependency_cols.append(new_dependency_col)
+
+        if 'singlefamilydetached' in project:
+            df = df[df['Dependency=Geometry Building Type RECS']
+                    == 'Single-Family Detached']
+
+        df = df.set_index(new_dependency_cols)
+        for col in list(df.columns.values):
+            if col in [count_col_label, weight_col_label]:
+                continue
+            df = df.rename(columns={col: 'Option={}'.format(col)})
+        df = df.sort_values(by=new_dependency_cols)
+        return df
+
+    def export_and_tag(self, df, filepath, project):
+        """
+        Add bottom-left script and source tag to dataframe (for non testing projects). Save dataframe to tsv file.
```

Comment:
Tony requested I leave it out because he is changing the format, and if I do it the new way it will fail integrity checks now, and if I do it the old way it will fail integrity checks later.

Reply:
Ok 👍